File: THCS/2023_2024/HaiDuong_2324/Cau4/Cau4.py
def count_valid_boxes(m, n):
    """Đếm số loại hộp bánh có thể mua."""
    count = 0
    for i in range(1, n + 1):
        total_packages = (i * (i + 1)) // 2  # Tính tổng số gói bánh
        if total_packages % m == 0:  # Kiểm tra chia đều cho m bạn
            count += 1
    return count

if __name__ == "__main__":
    # Nhập dữ liệu từ bàn phím
    m = int(input("Nhập số lượng học sinh (m): "))
    n = int(input("Nhập kích thước tối đa của hộp bánh (n): "))
    
    result = count_valid_boxes(m, n)
    print(result)  # In ra kết quả

File: THCS/2023_2024/HaiDuong_2324/Cau4/Cau4.cpp
#include <iostream>

using namespace std;

int countValidBoxes(int m, int n) {
    int count = 0; // Biến đếm số loại hộp bánh
    for (int i = 1; i <= n; i++) {
        // Tính tổng số gói bánh trong hộp
        int totalPackages = (i * (i + 1)) / 2; 
        // Kiểm tra xem tổng số gói bánh có chia đều cho m không
        if (totalPackages % m == 0) {
            count++;
        }
    }
    return count; // Trả về số loại hộp bánh
}

int main() {
    int m, n;
    
    // Nhập dữ liệu từ bàn phím
    cout << "Nhập số lượng học sinh (m): ";
    cin >> m;
    cout << "Nhập kích thước tối đa của hộp bánh (n): ";
    cin >> n;

    int result = countValidBoxes(m, n); // Tính số loại hộp bánh
    cout << result << endl; // In ra kết quả
    return 0;
}